src/drivers/general/queries/IrqlNotSaved/driver_snippet.c

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+// driver_snippet.c
+//
+
+#define SET_DISPATCH 1
+
+// Template. Not called in this test.
+void top_level_call() {}
+
+// Auxilliary function used in passing.
+// Note that even thoough this function doesn't annotate KIRQL,
+// it restores the IRQL correctly.
+VOID IrqlNotSaved_passAux(KIRQL outIrql) {
+    
+    KeRaiseIrql(KeGetCurrentIrql(), &outIrql);
+}
+
+// Passing case one.
+// Function directly calls a function that saves IRQL.
+VOID IrqlNotSaved_pass(_IRQL_saves_ KIRQL outIrql, PKSPIN_LOCK myLock) {
+
+    KeRaiseIrql(KeGetCurrentIrql(), &outIrql);
+
+}
+
+// Passing case two.
+// Function indirectly calls a function that saves IRQL.
+VOID IrqlNotSaved_pass2(_IRQL_saves_ KIRQL outIrql, PKSPIN_LOCK myLock) {
+
+    IrqlNotSaved_passAux(outIrql);
+
+}
+
+// Failing case one.
+// Function does nothing with IRQL.
+VOID IrqlNotSaved_fail1(_IRQL_saves_ KIRQL outIrql, PKSPIN_LOCK myLock) {
+
+}
+
+// Failing case two.
+// Function has a path where the IRQL is not saved.
+// This requies must-flow analysis.
```

Comment:
Wouldn't this test be passing for the current iteration of this rule?

Spelling: requires
Please use a spell checker. I like [this one](https://marketplace.visualstudio.com/items?itemName=EWoodruff.VisualStudioSpellCheckerVS2022andLater).

Reply:
Yes, this case does (incorrectly) pass.  It's named after the expected result, not the actual result.  

Thanks for the spell checker link, but this code is written in VS Code - I'll look to see if there's something similar for that.